docs/adding_packages/build_and_package.md

```diff
@@ -0,0 +1,36 @@
+# Build and Package
+
+This document gathers all the relevant information regarding the general lines to follow while writing either the `build()` or the `package()` methods.
+Both of these methods often take advantage of build helpers build binaries and install to the `package_folder`.
+
+<!-- toc -->
+## Contents
+
+  * [Build Method](#build-method)
+  * [Package](#package)<!-- endToc -->
+
+## Build Method
+
+* `build()` method should focus on build only, not installation. During the build, nothing should be written in `package` folder. Installation step should only occur in `package()` method.
+
+* The build itself should only rely on local files. Nothing should be downloaded from internet during this step. If external files are required, they should come from `requirements` or `build_requirements`, in addition to source files downloaded in `source()` or coming from recipe itself.
```

Comment:
if think if sources are different per platform, we do downloads in `build` method (e.g. android-ndk).

Reply:
Thats not the case - theres plenty of those and they work in `source()` --- maybe I misunderstood you?

Why would it not work?

Reply:
that's the case https://github.com/conan-io/conan-center-index/blob/master/recipes/android-ndk/all/conanfile.py#L57-L67

Reply:
Thats downloading pre compiled tooling which is why it's in the build method -- not the arch specific nature in this case